test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java

```diff
@@ -159,7 +160,8 @@ protected DiscoveryNode getDiscoveryNode(String id) {
         private volatile ReplicationTargets replicationTargets;
 
         private final PrimaryReplicaSyncer primaryReplicaSyncer = new PrimaryReplicaSyncer(
-            new TaskManager(Settings.EMPTY, threadPool, Collections.emptySet()),
+            new MockTransport().createTransportService(Settings.EMPTY, threadPool,
```

Comment:
Do we need to shut this (at least the threadpool) down cleanly?

Reply:
It's just using the fake transport so I don;t think the service needs any shutting down here. The `threadPool` is torn down via the parent `IndexShardTestCase` -> I think we're good?

Reply:
Oh right we already had a threadpool we're just using the transport service as a wrapper. Nvm.